devops/actions/run-tests/e2e/action.yml

```diff
@@ -24,26 +24,34 @@ runs:
   using: "composite"
   steps:
   - name: Checkout E2E tests
+    if: ${{ inputs.testing_mode != 'run-only'  || inputs.binaries_artifact }}
```

Comment:
nit: can we unify how we are checking `inputs.binaries_artifact`? here we just rely on non-empty being true but below we explicitly check it like `inputs.binaries_artifact != ''`, i have no opinion which one we do but would be a bit cleaner to always do the same thing